packages/flutter/test/widgets/modal_barrier_test.dart

```diff
@@ -114,7 +114,7 @@ void main() {
       reason: 'The route should have been dismissed by tapping the barrier.');
   });
 
-  testWidgets('ModalBarrier pops the Navigator when dismissed by primary tap down', (WidgetTester tester) async {
+  testWidgets('ModalBarrier pops the Navigator when dismissed by non-primary tap', (WidgetTester tester) async {
```

Comment:
There should probably be a test that verifies the dismiss happens on the up. It looks like this test would also pass when the dismiss happened on the down?

Reply:
I've changed the tests so that they verify that dismiss happens during the pointer up event